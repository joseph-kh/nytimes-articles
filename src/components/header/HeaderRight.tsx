import React from "react";
import { StyleSheet, TouchableOpacity } from "react-native";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import Row from "@/components/ui/containers/Row";
import { horizontalScale, moderateScale } from "@/utils/scaler";
import COLORS from "@/theme/colors";

const HeaderRight: React.FC = () => {
  return (
    <Row style={styles.container}>
      <TouchableOpacity>
        <MaterialCommunityIcons
          name="magnify"
          size={moderateScale(24)}
          color={COLORS.white}
        />
      </TouchableOpacity>
      <TouchableOpacity>
        <MaterialCommunityIcons
          name="dots-vertical"
          size={moderateScale(24)}
          color={COLORS.white}
        />
      </TouchableOpacity>
    </Row>
  );
};

const styles = StyleSheet.create({
  container: {
    justifyContent: "space-between",
    flexDirection: "row",
    width: moderateScale(90),
    paddingLeft: horizontalScale(20),
  },
});

export default HeaderRight;
