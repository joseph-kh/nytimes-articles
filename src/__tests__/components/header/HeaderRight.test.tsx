import React from "react";
import renderer from "react-test-renderer";
import HeaderRight from "@/components/header/HeaderRight";

describe("HeaderRight Component", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<HeaderRight />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
