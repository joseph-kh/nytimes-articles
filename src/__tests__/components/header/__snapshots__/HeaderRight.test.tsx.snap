// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderRight Component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingLeft": 40,
      "width": 135,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text />
  </View>
</View>
`;
